### ISSUE TITLE
A non-monotonic ROC is being produced by ggroc

### ISSUE THREAD

**Issue by AngCamp** [xrobin/pROC#121]
I have trained a random forest with 10-fold cross validation using the randomforest package.  I withhold the data, train the model for each fold then test it using the randomforest predict function and give that to  the roc() function.  I create a list of these roc objects for ggroc and then I give it to the ggroc function.

The list of roc objects is [here](https://drive.google.com/drive/folders/1ZYs1gMzPr64lV7WQmVOu1zt0YSVak9cr?usp=drive_link)
It is the `my_roclist.rds`.

```
# libraries needed (well maybe not ggplot)
library(pROC)
library(ggplot2)

# load the my_roclist.rds file I've provided in the google drive
hochgerner.roc.list <- readRDS("my_roclist.rds")
  
# plotting
jpeg('hochgerner2018_earlysig_roc.jpeg', width = 600, height=600)
ggroc(hochgerner.roc.list,  alpha = 1,
      colour = "red", linetype = 'solid',
      size = 4, legacy.axes = TRUE) + 
  theme_classic() + 
  ggtitle("Hochgerner et al., (2018) ROC") + 
  xlab("FPR") + ylab("TPR") + 
  geom_segment(aes(x = 0, xend = 1, y = 0, yend = 1), linewidth = 2,
               color="darkgrey", linetype="dashed") +
  theme(plot.title = element_text(size = 20, face = "bold", hjust = 0.5),
        axis.text=element_text(size=20, face = "bold", colour="black"),
        axis.title =element_text(size=20, face = "bold", colour="black"),
        panel.border = element_rect(color = "black",
                                    fill = NA,
                                    linewidth = 2),
        plot.margin = margin(t = 10, r = 10, b = 10, l = 10) ) +
  scale_x_continuous(expand = expansion(mult = c(0, 0)), breaks = c(0,0.25,0.5,0.75,1)) + 
  scale_y_continuous(expand = expansion(mult = c(0, 0)),  breaks = c(0.25,0.5,0.75,1))
dev.off()
```

![image](https://github.com/xrobin/pROC/assets/77176433/b257a48f-fefc-492e-9550-77337a7c0cd4)

**Comment by xrobin** [comment]
Thanks for the bug report and the code to reproduce it.

It  turns out it can be simplified to 

    ggroc(hochgerner.roc.list,  colour = "red")

As you set a static colour, all the curves have the same aesthetics and are bunched up together as a single one.

Setting the "group" aes solves the issue

    ggroc(hochgerner.roc.list,  colour = "red", aes="group")

This should probably be set by default. I'll see if that can be done.

**Comment by xrobin** [comment]
This is now fixed in the develop branch.  You can install it with `devtools::install_github("xrobin/pROC@develop")`.

I will aim to release this fix soon but I'd like to include #120 too.

Until then please use the workaround described above.
